apps/tailwind-components/components/form/Fields.vue

```diff
@@ -185,63 +51,44 @@ watch(
 );
 
 onMounted(() => {
-  updateRowKey();
   copyConstantValuesToModelValue();
 });
 </script>
 
 <template>
-  <div>
-    <template v-for="(column, index) in metadata.columns">
-      <div
-        id="_scroll_to_top"
-        v-if="index === 0 && column.columnType !== 'HEADING'"
-        class="-mt-10"
-      >
-        <span v-intersection-observer="onIntersectionObserver"></span>
-      </div>
+  <div ref="container">
+    <template v-for="column in columns">
       <div
         v-if="column.columnType === 'HEADING'"
-        :id="column.id"
-        v-intersection-observer="onIntersectionObserver"
+        :id="`${column.id}-form-field`"
+        v-intersection-observer="[onIntersectionObserver, { root: container }]"
       >
         <h2
           class="first:pt-0 pt-10 font-display md:text-heading-5xl text-heading-5xl text-form-header pb-8"
-          v-if="column.id !== '_scroll_to_top' && visibleMap[column.id]"
+          v-if="column.label"
         >
           {{ column.label }}
         </h2>
       </div>
-
-      <div
-        style="height: 100px"
-        v-on-first-view="() => checkVisibleExpression(column)"
-        v-else-if="
-          !column.id.startsWith('mg_') && visibleMap[column.id] === undefined
-        "
-      ></div>
       <FormField
         class="pb-8"
-        v-else-if="
-          visibleMap[column.id] === true &&
-          !Object.keys(constantValues || {}).includes(column.id)
-        "
+        v-else-if="!Object.keys(constantValues || {}).includes(column.id)"
+        v-intersection-observer="[onIntersectionObserver, { root: container }]"
         v-model="modelValue[column.id]"
         :id="`${column.id}-form-field`"
         :type="column.columnType"
         :label="column.label"
         :description="column.description"
         :rowKey="rowKey"
         :required="isRequired(column.required ?? false)"
-        :error-message="errors[column.id]"
-        :ref-schema-id="column.refSchemaId || schemaId"
+        :error-message="errorMap[column.id]"
+        :ref-schema-id="column.refSchemaId"
         :ref-table-id="column.refTableId"
         :ref-label="column.refLabel || column.refLabelDefault"
         :ref-back-id="column.refBackId"
-        :invalid="errors[column.id]?.length > 0"
-        @update:modelValue="onUpdate(column, $event ?? '')"
-        @blur="onBlur(column)"
-        @focus="onFocus(column)"
+        :invalid="errorMap[column.id]?.length > 0"
+        @update:modelValue="emit('update', column)"
```

Comment:
Would this cause the validation to occur before the user finished entering a value?

Reply:
as discussed, desired behavior to remove a error if user has it